src/dom.ts

```diff
@@ -200,29 +200,56 @@ export function containsClass(element: HTMLElement, className: string): boolean
  * @example
  * var fragment = dom.fromString('<tr>');
  */
-export function fromString(html: string): DocumentFragment {
-	html = String(html);
+export function fromString(html: string): DocumentFragment  {
+	let fragment: DocumentFragment;
+	let useContextualFragment = has('dom-contextual-fragment');
 
-	const match = html.match(/<\s*([\w\:]+)/);
-	const tag = match ? match[1].toLowerCase() : '';
 	let master: HTMLElement = document.createElement('div');
```

Comment:
This in theory shouldn't require casting anymore to `HTMLElement`.  Current versions of TypeScript create the appropriate type now, using a string literal.